src/field/overlay_01_021E6880.c

```diff
@@ -0,0 +1,1192 @@
+            fieldSystem->lastTouchMenuInput = 0;
+        } else if ((newKeys & PAD_BUTTON_X) || fieldSystem->lastTouchMenuInput != 0) {
+            if (ov01_021F6B00(fieldSystem) == 4) {
+                MenuInputStateMgr_SetState(&fieldSystem->menuInputState, MENU_INPUT_STATE_BUTTONS);
+                input->wantToReloadTouchscreenMenu = TRUE;
+                input->wantToOpenMenu = TRUE;
+            } else if (hasBag) {
+                input->wantToOpenMenu = TRUE;
+                if (fieldSystem->lastTouchMenuInput == 1) {
+                    fieldSystem->lastTouchMenuInput = 0;
+                }
+            }
+        } else if (newKeys & PAD_BUTTON_A) {
+            input->wantToInteract = TRUE;
+        }
+
+        if (heldKeys & (PAD_KEY_UP | PAD_KEY_DOWN | PAD_KEY_LEFT | PAD_KEY_RIGHT)) {
+            input->unk0_5 = TRUE;
+            input->unk0_6 = TRUE;
+        }
+        input->unk0_8 = TRUE;
+    } else {
+        fieldSystem->lastTouchMenuInput = 0;
+    }
+
+    if (avatar14 == 3 && avatar10 == 1) {
+        input->unk0_7 = TRUE;
+    }
+    if (avatar14 == 3) {
+        input->unk0_1 = TRUE;
+    }
+
+    if ((direction == DIR_NORTH && heldKeys & PAD_KEY_UP) || (direction == DIR_SOUTH && heldKeys & PAD_KEY_DOWN) || (direction == DIR_WEST && heldKeys & PAD_KEY_LEFT) || (direction == DIR_EAST && heldKeys & PAD_KEY_RIGHT)) {
+        input->currentMovingDirectionMaybe = direction;
+    } else {
+        input->currentMovingDirectionMaybe = DIR_NONE;
+    }
+
+    input->nextMovingDirectionMaybe = sub_0205DD94(fieldSystem->playerAvatar, newKeys, heldKeys);
+}
+
+int FieldInput_Process(FieldInput *input, FieldSystem *fieldSystem) {
+    if (!input->unk0_13 && TryStartMapScriptByType(fieldSystem, 1) == TRUE) {
+        return 1;
+    }
+
+    if (!input->unk0_13) {
+        BOOL unkBool = HasEnoughAlivePokemonForDoubleBattle(SaveArray_Party_Get(fieldSystem->saveData));
+        if (Save_VarsFlags_CheckHaveFollower(Save_VarsFlags_Get(fieldSystem->saveData)) == TRUE) {
+            unkBool = TRUE;
+        }
+        if (TryGetSeenByNpcTrainers(fieldSystem, unkBool) == TRUE) {
+            sub_0205CF44(fieldSystem->playerAvatar);
+            MapObjectManager_PauseAllMovement(fieldSystem->mapObjectManager);
+            if (FollowMon_IsActive(fieldSystem)) {
+                MapObject_UnpauseMovement(FollowMon_GetMapObject(fieldSystem));
+            }
+            return 1;
+        }
+    }
+
+    if (input->unk0_7) {
+        ClearFlag965(Save_VarsFlags_Get(fieldSystem->saveData));
+        if (ov01_021E7628(fieldSystem) == TRUE) {
+            return 1;
+        }
+    }
+
+    if (BugContestTimeoutCheck(fieldSystem)) {
+        return 1;
+    }
+
+    u32 flags = 0;
+    int direction = sub_0205DD94(fieldSystem->playerAvatar, input->newKeys, input->heldKeys);
+    if (StrengthFlagAction(Save_VarsFlags_Get(fieldSystem->saveData), FLAG_ACTION_CHECK)) {
+        flags |= (1 << 0);
```

Comment:
These look like flags for whenever HMs that only need to be activated once per map are active, if you'd like to name them